# Project X -- [Prisha] [Joshi]

**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [1]:
# code here
library(data.table)
DF <- fread("/anvil/projects/tdm/data/death_records/DeathRecords.csv")
head(DF)
tapply(DF$Age , DF$Sex , mean , na.rm = TRUE)

Id ResidentStatus Education1989Revision Education2003Revision
1 1  1              0                     2                    
2 2  1              0                     2                    
3 3  1              0                     7                    
4 4  1              0                     6                    
5 5  1              0                     3                    
6 6  1              0                     5                    
  EducationReportingFlag MonthOfDeath Sex AgeType Age AgeSubstitutionFlag ...
1 1                      1            M   1       87  0                   ...
2 1                      1            M   1       58  0                   ...
3 1                      1            F   1       75  0                   ...
4 1                      1            M   1       74  0                   ...
5 1                      1            M   1       64  0                   ...
6 1                      1            F   1       93  0                   ...
  CauseRecode39 NumberOfEntityAxisConditions NumberOfRecordAxisConditions Race
1 24            1                            1                            1   
2 21            3                            3                            1   
3 28            2                            2                            1   
4 22            3                            3                            1   
5 21            1                            1                            1   
6 37            5                            5                            1   
  BridgedRaceFlag RaceImputationFlag RaceRecode3 RaceRecode5 HispanicOrigin
1 0               0                  1           1           100           
2 0               0                  1           1           100           
3 0               0                  1           1           100           
4 0               0                  1           1           100           
5 0               0                  1           1           100           
6 0               0                  1           1           100           
  HispanicOriginRaceRecode
1 6                       
2 6                       
3 6                       
4 6                       
5 6                       
6 6

F        M 
76.67668 70.22786

the average age of death for females is higher than males. males have an average death age of 70 whereas females live till 77 

## Question 2

In [1]:
# code here
library(data.table)
df <- fread("/anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv")
#tapply(df$SPEND , df$PRODUCT_NUM, sum)
a <- sort(tapply(df$SPEND , df$PRODUCT_NUM, sum), decreasing = TRUE )
head(a,10)



8511   1367192     85311     74424    766108     85201   4889358   1344763 
131399.78 111433.78 102928.59  75787.49  66085.00  65605.34  63823.61  58170.84 
     8523     89415 
 53845.65  50032.42

the top 10 types of product numbers according to the total amount of money spent on it are displayed in descending order. the highest money that was spent was on product number 8511 which was 131399.78. the lowest product number was 89415 with the amount of 50032.42

## Question 3

In [4]:
# code here
library(data.table)
DF1 <- fread("/anvil/projects/tdm/data/beer/reviews_sample.csv")
a<-tapply( DF1$score , DF1$date, mean , na.rm= TRUE)
sort(a)
tail(sort(a), 3)


2002-02-13 2002-05-10 2001-10-24 2001-12-23 2001-04-25 2002-10-22 2002-08-18 
  1.000000   1.900000   2.000000   2.000000   2.080000   2.353333   2.422500 
2001-08-17 2001-09-25 2002-05-13 2002-08-09 2001-10-16 2001-10-18 2002-02-28 
  2.530000   2.560000   2.575000   2.593333   2.600000   2.600000   2.700000 
2002-04-08 2002-02-15 2002-09-11 1998-09-06 2002-08-16 1998-04-16 1998-02-19 
  2.735000   2.760000   2.760000   2.770000   2.805000   2.850000   2.855000 
2001-12-14 2003-07-01 2002-03-10 2002-02-24 2004-08-18 2002-08-24 2002-06-22 
  2.860000   2.900000   2.910000   2.930000   2.970000   2.975000   3.040000 
2006-03-10 2001-06-24 1998-07-07 2004-10-06 2003-04-19 2002-02-18 2002-05-26 
  3.053000   3.080000   3.100000   3.106000   3.112500   3.120000   3.136667 
2002-07-20 2004-07-14 2002-08-13 2007-05-03 2003-12-31 2002-10-26 2005-01-19 
  3.140000   3.162857   3.166667   3.169091   3.187778   3.190000   3.191250 
1999-02-09 2003-08-03 2001-10-20 2002-03-02 2002-06-15 2003-07-24 2007-07-31 
  3.195000   3.201000   3.210000   3.230000   3.231667   3.238333   3.247500 
2005-11-18 2002-03-23 2006-05-14 2004-03-14 2002-08-19 2005-07-24 2003-06-04 
  3.250714   3.260000   3.266923   3.271000   3.280000   3.282727   3.286000 
2002-11-19 2002-10-16 2006-06-09 2002-09-06 2006-08-08 2002-07-24 2002-03-25 
  3.295000   3.300000   3.304545   3.305000   3.306250   3.315000   3.320000 
2003-09-29 2002-04-05 2004-10-01 2005-04-08 2006-11-28 2002-02-03 2002-09-04 
  3.328000   3.336000   3.343636   3.347143   3.347368   3.350000   3.350000 
2004-11-13 2007-06-29 2004-01-20 2003-01-27 2006-09-12 2003-08-17 2004-08-02 
  3.350000   3.356842   3.358571   3.360000   3.362632   3.365000   3.365714 
2002-11-10 2003-08-30 2007-05-09 2003-09-12 2004-04-02 2007-11-18 2002-12-20 
  3.375000   3.375833   3.376250   3.377000   3.380000   3.380385   3.385000 
2004-03-01 2004-07-04 2003-07-20 2004-07-10 2003-07-29 2007-09-12 2000-12-30 
  3.387500   3.388000   3.392000   3.392857   3.395000   3.396667   3.400000 
2001-07-11 2002-10-12 1998-05-25 2001-11-30 2003-04-10 2005-02-25 2001-12-20 
  3.400000   3.400000   3.400000   3.400000   3.403333   3.406667   3.410000 
2003-07-17 2003-12-06 2004-10-05 2006-04-17 2002-08-07 2007-03-30 2004-12-15 
  3.410000   3.410833   3.411111   3.413000   3.413333   3.416364   3.418000 
2002-06-29 2004-03-17 2003-08-23 2002-10-23 2007-08-11 2002-11-03 2004-11-03 
  3.420000   3.420000   3.421111   3.423333   3.425882   3.426000   3.428182 
2005-08-02 2007-06-11 2007-06-27 2003-06-26 2005-06-13 2005-09-28 2002-04-22 
  3.428889   3.431538   3.435714   3.436250   3.439000   3.439000   3.440000 
2003-03-16 2002-10-03 2003-03-21 2005-05-04 2003-12-30 2004-06-29 2005-06-01 
  3.440000   3.442000   3.442500   3.442500   3.442727   3.442857   3.442857 
2005-05-27 2002-09-24 2002-08-23 2004-06-13 2002-11-28 2005-04-29 2002-10-11 
  3.443077   3.443333   3.444000   3.445385   3.446667   3.447143   3.448333 
2004-12-17 1998-02-21 1998-03-09 2004-03-18 2007-08-10 2004-02-14 2004-04-01 
  3.449091   3.450000   3.450000   3.450000   3.453000   3.453333   3.453636 
2004-03-22 2003-07-27 2004-11-20 2002-08-01 2002-04-19 2003-06-14 2007-04-24 
  3.454545   3.456667   3.456667   3.457500   3.458000   3.458000   3.458235 
2005-07-21 2006-03-20 2003-08-09 2002-06-16 2007-04-23 2007-12-06 2003-01-07 
  3.460000   3.460000   3.461111   3.462500   3.463333   3.463333   3.468333 
2003-09-21 2005-05-16 2004-12-20 2007-09-04 2003-03-28 2002-08-02 2002-11-05 
  3.470000   3.471176   3.471250   3.471667   3.472222   3.473333   3.473333 
2003-11-09 2007-07-02 2001-10-03 2001-10-31 2002-06-26 2005-02-13 2005-07-15 
  3.475000   3.477368   3.480000   3.480000   3.480000   3.480000   3.484545 
2002-09-15 2006-05-29 2002-04-13 2004-11-27 2005-02-18 2004-11-10 2004-06-22 
  3.485714   3.485789   3.486667   3.486667   3.487500   3.488333   3.489167 
2004-03-21 2005-09-12 2007-01-17 2007-02-22 2003-06-25 2003-04-08 2005-08-29 
  3.491667   3.492308 

2001-04-26 2001-06-18 2002-01-26 
         5          5          5

the three dates with the perfect mean score for beer reviews of 5 are 26th april 2001 , 18th july 2001, and 26th january 2002.

## Question 4

In [10]:
# code here
library(data.table)
df2 <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
names(df2) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
merge <- paste(df2$CITY, df2$STATE, sep=",")
z<-tapply(df2$TRANSACTION_AMT , merge , sum)
w <- sort (z,decreasing = TRUE)
print(head(w, 20))

               ,      NEW YORK,NY       HOUSTON,TX        DALLAS,TX 
        17299729         11345027          7606806          4748262 
  LOS ANGELES,CA    WASHINGTON,DC       CHICAGO,IL SAN FRANCISCO,CA 
         4569952          4273606          3179470          2061441 
BEVERLY HILLS,CA       ATLANTA,GA OKLAHOMA CITY,OK      ST LOUIS,MO 
         2053148          1892356          1751431          1639570 
  SAN ANTONIO,TX         MIAMI,FL         TULSA,OK     GREENWICH,CT 
         1583292          1541867          1341956          1340112 
   WASHINGTOM,DC       MIDLAND,TX        DENVER,CO    CINCINNATI,OH 
         1300341          1210584          1183471          1096983 


the top 20 city state pairs that donated themost amount of money for the 1980 elections are dispalyed . the unknown city state pair that donated the most amount was of 17299729 and the least was by cincinnati with an amount of 1096983

## Question 5

In [1]:
# code here
library(data.table)
df3<- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")
date <- paste(df3$Month, df3$DayofMonth , df3$Year ,sep="/")
d <-tapply(df3$DepDelay, date, mean, na.rm = TRUE)
p<-sort(d)
print(tail(p, 6))



12/20/1990  2/15/1990  2/16/1990 12/28/1990 12/22/1990 12/21/1990 
  27.34540   28.12302   36.19422   43.91443   45.22225   45.66178 


the worst dates with the largest mean departure delay in the year 1990 are as follows. the highest mean time was on 21st december with 45.66178 hrs and the lowest 20th december with 27.34540. we could infer that this delay might be due to weather or the holidays 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.